### DISCUSSION TITLE
several instances of netbox on the same server

### DISCUSSION THREAD

**Discussion by quentindavid** [netbox-community/netbox#6575] [Q&A] [answered]
Hi, I have 2 instances of netbox on a server. It seems rack information is mixed up since last update (I updated from an old version to the latest).

Is it possible that information mix-up between the two instances due to the use of redis ?
Is there a way to prevent this ?

Or maybe it's only a cache issue ?

Example of what I get : when I list racks, I can see racks of the other instance of netbox, and when I click if the ID is existing it's showing the corresponding ID, but if it does not it throws an error "this id not exist".

**Comment by candlerb** [comment, marked as answer]
> Is it possible that information mix-up between the two instances due to the use of redis ?

Yes, if they are both pointing to the same redis instance (port and database number).

> Or maybe it's only a cache issue ?

Caching is what you see, but it will also mix up background jobs like webhooks which also are queued via redis.

You should point your second netbox instance to different [database numbers](https://redis.io/commands/select) within redis.

```
REDIS = {
    'tasks': {
        ...
        'DATABASE': 0,  # change to 2
    },
    'caching': {
        ...
        'DATABASE': 1,  # change to 3
    }
```

**Reply by quentindavid** [reply]
Ok I understand ! It seems to work much better now ! Thanks a lot :)
For the instance where I use the new database number it seems fine, but how can I reset the databases 1 and 2 ?

**Reply by candlerb** [reply]
Caches expire after 15 minutes anyway.  You can use redis-cli and the [flushdb](https://redis.io/commands/flushdb) command; or netbox manage.py has a command to invalidate caches.

Note that each instance uses *two* databases, and by default these are 0 and 1, not 1 and 2.

**Reply by quentindavid** [reply]
Yes, my bad, it's fine then it looks good now :)
and yes it's 0/1 and 2/3.

Thanks a lot for the explanations ! Everything seems fine now !
